src/UnitTest/test_quote_prov.cpp

```diff
@@ -492,9 +501,18 @@ static void GetCrlTestICXV3()
 static inline void VerifyCollateral(sgx_ql_qve_collateral_t* collateral)
 {
     boolean TEST_SUCCESS = false;
-    ASSERT_TRUE(collateral != nullptr);
+    std::string collateral_version =
+        get_env_variable(ENV_AZDCAP_COLLATERAL_VER);
 
-    ASSERT_TRUE(collateral->version == 1);
+    ASSERT_TRUE(collateral != nullptr);
+    if (!collateral_version.compare("v1") || !collateral_version.compare("v2"))
+    {
+        ASSERT_TRUE(collateral->version == 1);
+    }
+    else
+    {
+        ASSERT_TRUE(collateral->version == 3);
```

Comment:
suggestion to make it an explicit check for comparing "v3", as there will be  version 4 etc and we don't want to default to 3 regardless.

Reply:
Currently in the code the only values possible are v1/v2/v3 as the version defaults to V3 if no environment variable specified. If an incorrect environment variable is specified - it will default to v3. When we introduce code for v4 - we would have to make explicit check and assign values for v4 at this spot.

Reply:
Just a suggestion, i've always preferred to be explicit because we may not be the ones making the change in the future and it will need to potentially be explicit for "v3" then anyway when there is a "v4" was my thought. Your call.